src/java.desktop/macosx/native/libawt_lwawt/awt/CTextPipe.m

```diff
@@ -597,20 +598,23 @@ render using the right font (see bug 7183516) when attribString is not
     {
         jchar unichars[len];
         (*env)->GetStringRegion(env, str, 0, len, unichars);
-        JNF_CHECK_AND_RETHROW_EXCEPTION(env);
+        CHECK_EXCEPTION();
 
         // Draw the text context
         DrawTextContext(env, qsdo, awtStrike, unichars, len, x, y);
     }
     else
     {
         // Get string to draw and the length
-        const jchar *unichars = JNFGetStringUTF16UniChars(env, str);
+        const jchar *unichars = (*env)->GetStringChars(env, str, NULL);
+        if (unichars == NULL) {
+            return;
```

Comment:
Do not we need to throw an exception here? Otherwise, GetStringChars error will be ignored?

Reply:
> The changes look good to me, but please see my comment from my review about documenting `NormalizedPathNSStringFromJavaString()` API.

I see it. I will copy what you wrote in there.

Reply:
> I read it and not sure that it is fine to ignore this error, why not throw an exception and signal the CTextPipe_doDrawString that an error occurred like InvalidPipeException or something(Sometimes we wrap other exception like OOM into the InvalidPipeException and this seems similar case)?

OK. I will do something like throw OOME or InternalError. Since we already checked for NULL as an arg any failure here implies a deep VM problem rather than anytjhing like a 2D invalid pipe